workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogOverlay.kt

```diff
@@ -6,23 +6,28 @@ import com.squareup.workflow1.ui.R
 import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
 
 /**
- * Returns the most recent [Overlay] rendering [shown][OverlayDialogHolder.show] in this [Dialog],
- * or throws a [NullPointerException] if the receiver was not created via
+ * Returns the most recent [Overlay] rendering [shown][OverlayDialogHolder.show]
+ * in this [Dialog]. Throws a [NullPointerException] if the receiver was not created via
  * [OverlayDialogFactory.buildDialog].
+ *
+ * Note that it is not safe to call this until after the [Dialog] has been shown.
```

Comment:
Can't we add this hint to the exception message below?

Reply:
That exception is about trying to read `overlay` before having set it, not about the `decorView` nightmare itself. This is just the usual `fooOrNull` / `foo` pairing, where calling `foo` means you're asserting that you have previously set `foo`.